File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let result = 0;
    
    if(s[0]===')') return false;
    
    for(let c of s) {
        if(result<0) return false;
        
        if(c==='(') result++;
        if(c===')') result--;
    }
    
    return !result;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = []

    for c in s:
        if c == '(':
            answer.append(c)
        else:
            if len(answer) > 0 and answer[-1] == '(':
                answer.pop()
            else:
                answer.append(c)

    return True if len(answer) == 0 else False